solution1.py

```diff
@@ -0,0 +1,39 @@
+# // Time Complexity : o(n^2) for deepcopy at every node
+# // Space Complexity : o(n*height) ==> o(nlogn) for complete binary OR o(n^2) for one linked-list tree
```

Comment:
The space will be O(n^2) for any scenario because n lists are created n total.
n nodes - n lists, n*n or n^2.